src/process_utility.c

```diff
@@ -283,9 +283,14 @@ foreach_chunk_relid(Oid relid, process_chunk_t process_chunk, void *arg)
 	int			ret;
 
 	if (NULL == ht)
+	{
+		cache_release(hcache);
 		return -1;
+	}
 
+	hcache->release_on_commit = false;
```

Comment:
Is this correct when the `process_chunk` handler function isn't doing cross-commit operations? Why not put this in `foreach_chunk` in that case?

Reply:
Moved this logic down to the vacuum handling. Now more self-contained